### DISCUSSION TITLE
Autonomous flight activation

### DISCUSSION THREAD

**Discussion by alexandraH99** [bitcraze/discussions#106] [Q&A]
Hi!

I wrote a controller basically just makes the the crazyflie hover in the air without any input from joysticks. However,  I'm confused by how to activate the crazyflie to hover and stop. What should I do so that once I start the Crazyflie, it starts to hover after a few seconds, and then it can stop with some sort of command?

Thanks!

**Comment by tobbeanton** [comment]
Could you please share a bit more information? How are you controlling it, python library? You write "once I start the Crazyflie", does that mean power it on? You could checkout the [app layer framework](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/app_layer/).

**Reply by alexandraH99** [reply]
Hi,

- I'm basically following the format of the push demo and wall following demo in the bitcraze firmware repository, like compiling a c file to control the crazyflie I think. However, the crazyflie doesn't seem to do anything. It's not even doing the debug printing right at the beginning of the while(1) loop.
- Yes that means power it on.

Thanks.

**Reply by krichardsson** [reply]
> It's not even doing the debug printing 

It sounds like you possibly did not flash the correct binary? I would suggest that you start by building and flashing one of the examples first, for instance `examples/app_hello_world`, just to make sure you get some logging through.

**Reply by alexandraH99** [reply]
Hi,

Before testing my own program, I tried most of those examples, and they all worked fine. However, I'm still having trouble with my program.

Thanks.

**Reply by alexandraH99** [reply]
Hi, the debug_print problem is solved. However, I'm still confused by how to control the crazyflie to start and stop with some sort of commands during autonomous flight, and without any deck. Thanks.

**Comment by knmcguire** [comment]
Let's start a new thread for this:
> Hi, the debug_print problem is solved. However, I'm still confused by how to control the crazyflie to start and stop with some sort of commands during autonomous flight, and without any deck. Thanks.

If you are using the app layer, you should probably make an parameter group in the c file that you made your app-layer application in. Please read up here: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/logparam/

If you would add a value (probably PARAM_UINT8 would be fine) called for instance 'startFlight' and recompile & flash the app again, you would able to change that value from the cfclient or the cflib. Then you just have to make sure that your logic in the app layer takes that parameter change and does something with it (like taking off and landing).

**Reply by alexandraH99** [reply]
The problem is solved. Thanks!
